### PR TITLE
[#339] Support GHC-8.8

### LINKED ISSUE #339: Support GHC-8.8
- [ ] Change default GHC
- [ ] Add and change default stack LTS

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #339

* Update CHANGELOG
    - [x] [summoner-cli/CHANGELOG](https://github.com/kowainik/summoner/blob/master/summoner-cli/CHANGELOG.md).
    - [ ] [summoner-tui/CHANGELOG](https://github.com/kowainik/summoner/blob/master/summoner-tui/CHANGELOG.md).
- [x] Keep code style used in the changed files (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] Use the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [x] Update documentation (README, haddock) if required.
- [ ] Create a new test project using `summoner` and check that the changes work as expected.

### CODE REVIEW

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 1]
```diff
@@ -211,6 +211,8 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
             -Wmissing-export-lists
             -Wpartial-fields
             |]
+        <> memptyIfFalse (settingsTestedVersions `hasLeast` Ghc881)
+            "-Wmissing-deriving-strategies\n"
```
Comment (vrom911) [inline summoner-cli/src/Summoner/Template/Cabal.hs]: Should it be error?

Comment (vrom911) [reply summoner-cli/src/Summoner/Template/Cabal.hs]: Also, we should definitely go for the conditional options per GHC versions as you suggested before!

File: README.md [hunk 2]
```diff
@@ -588,7 +589,7 @@ If you'd like to take part in the development processes, here are a few things t
 * Summoner is a multi-package project which has two packages inside: `summoner-cli` and `summoner-tui`. We are supporting the same functionality in both of them, so, if it's possible, you should add/change the code in both of the packages.
 * Summoner uses the custom prelude [`relude`](https://github.com/kowainik/relude).
 * We are using the [Kowainik style guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide).
-* To sum up, [here is the Contributing guide](https://github.com/kowainik/org/blob/master/CONTRIBUTING.md#contributing-to-the-kowainik-repositories) we use across the repositories.
+* To sum up, [here is the Contributing guide](https://github.com/kowainik/.github/blob/master/CONTRIBUTING.md#contributing-to-the-kowainik-repositories) we use across the repositories.
```
Comment (chshersh) [inline README.md]: Thanks for noticing to update this thing!

File: summoner-cli/src/Summoner/Template/Cabal.hs (hunk 1 above)
Comment (chshersh) [reply summoner-cli/src/Summoner/Template/Cabal.hs]: I think it should be error. It's an interesting question, to which warnings we should add `-Werror` but I'd love to add it here since strategies are implied by our style-guide and this is a good default for our projects. And yes, I agree that after we introduce common stanzas, a lot of this stuff will be improved 🙂  Thanks for support my idea!
